src/components/Popup/Popup.tsx

```diff
@@ -146,11 +183,32 @@ export default class Popup extends UIComponent<Extendable<IPopupProps>, IPopupSt
         basic={basic}
         {...rtl && { dir: 'rtl' }}
         styles={{ root: computedStyle }}
+        {...accessibility.attributes.popup}
+        {...accessibility.keyHandlers.popup}
       >
         {content}
       </Popup.Content>
     )
   }
 
+  private onTriggerClick = () => {
+    this.setState({ popupOpened: !this.state.popupOpened })
+  }
+
+  private openPopup = (e: Event, afterRenderCb?: () => void) => {
+    e.preventDefault()
+    this.setState({ popupOpened: true }, () => afterRenderCb && afterRenderCb())
```

Comment:
no need for `afterRenderCb` logic, just pass the parameter along, it will be handled by `setState` function
```tsx
this.setState({ popupOpened: false }, afterRenderCb)
```
same below

also, `openPopup` and `closePopup` contain duplicated logic; rather have one method with `handlePopupState` that contains `popupOpened` as param;

```typescript
private handlePopupState = (e: Event, popupOpened: boolean, afterRenderCb?: () => void) => {
    e.preventDefault()
    this.setState({ popupOpened }, afterRenderCb)
}
```

Reply:
ok, make sense. I thought it would bring more readability :)